server/flb_manager.ts

```diff
@@ -22,32 +18,28 @@ interface FlbManager {
 interface FlbInstance {
   connect: (socket: Socket) => void
   disconnect: (socket: Socket) => void
-  write: (data: string) => void
   disconnectAll: () => void
   connectedCount: () => number
   datasource: () => string
 }
 
-function getToken() {
-  if (process.env.VIVO_TOKEN) {
-    console.log("Using VIVO_TOKEN variable for token.")
-    return process.env.VIVO_TOKEN
-  }
-  console.log("Generating token as no VIVO_TOKEN variable found.")
-  return crypto.randomBytes(32).toString('hex')
-};
-
 function spawnFluentBit({ datasource }: FluentBitOpts): FlbInstance {
   const sockets = new Set<Socket>()
 
-  const ds = datasource === 'http' ? 'stdin' : datasource
-
-  const flb = cp.spawn(FLUENT_BIT_EXEC_PATH, [
-    '-i', ds,
+  const args = [
+    '-i', datasource,
+  ].concat(datasource === 'http' ? [
+    `-phost=${FLUENT_BIT_HTTP_HOST}`,
+    `-pport=${FLUENT_BIT_HTTP_PORT}`,
+  ] : []).concat([
     '-o', 'stdout',
     '-p', 'format=json',
     '-f', '0.2'
   ])
+
+  const flb = cp.spawn(FLUENT_BIT_EXEC_PATH, args, {
+    stdio: ['ignore', 'pipe', 'pipe']
```

Comment:
To clarify, "ignore" means fluent-bit stdin will be redirected to `/dev/null` @edsiper